projects/demo/src/app/highlighter.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { codeToHtml } from 'shiki';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HighlighterService {
+  async highlight(code: string, lang: string): Promise<string> {
+    return codeToHtml(code.toString().trim(), {
+      lang,
+      theme: 'catppuccin-frappe'
```

Comment:
ultra-nit: I prefer a white or light theme because I'm daltonian, but it's not a big deal.